hrabsence/CRM/HRAbsence/Form/AbsenceRequest.php

```diff
@@ -517,320 +541,410 @@ static function formRule($fields, $files, $self) {
+          $mailprm[$val]['display_name'] = $managerContactResult['values'][$val]['display_name'];
+          $mailprm[$val]['email'] = $managerContactResult['values'][$val]['email'];
         }
-        self::sendAbsenceMail($mailprm, $sendTemplateParams);
       }
     }
-    else {
-      if (CRM_Utils_Request::retrieve('aid', 'Positive', $this)) {
-        $activityIDs = CRM_Utils_Request::retrieve('aid', 'Positive', $this);
-      }
 
-      if (array_key_exists('_qf_AbsenceRequest_done_cancelabsence', $submitValues)) {
-        $statusId = CRM_Utils_Array::key('Cancelled', $activityStatus);
-        $statusMsg = ts('Absence(s) have been Cancelled');
-      }
-      elseif (array_key_exists('_qf_AbsenceRequest_done_cancel', $submitValues)) {
-        $session->pushUserContext(CRM_Utils_System::url('civicrm/absences', "reset=1&cid={$this->_targetContactID}", false, 'hrabsence/list'));
-      }
-      civicrm_api3('Activity', 'create', array(
+    return $mailprm;
+  }
+
+  /**
+   * Updates acivities to given status for current leave request.
+   * 
+   * @param type $status
+   *   Status to which activities should be set
+   * 
+   * @return array
+   *   Data of main activity for curent leave request
+   */
+  private function updateActivitiesStatus($status) {
+    $statusId = CRM_Utils_Array::key($status, $this->_activityStatuses);
+    $result = civicrm_api3('Activity', 'create', 
+      $activityParam = [
+        'sequential' => 1,
         'id' => $this->_activityId,
         'activity_type_id' => $this->_activityTypeID,
         'status_id' => $statusId
+      ]
+    );
+    $subact = civicrm_api3('Activity', 'get', ['return' => "id", 'source_record_id' => $result['id']]);
+    foreach($subact['values'] as $val) {
+      civicrm_api3('Activity', 'create', ['id' => $val['id'] , 'status_id' => $statusId]);
+    }
+    
+    return $result;
+  }
+
+  /**
+   * Sets leave request as approved by changing the main activity's status and
+   * its subactivities to Completed.
+   * 
+   * @param array $absentDateDurations
+   *   List of durations (days) of leave
+   * 
+   * @return array
+   *   Data of main activity for curent leave request
+   */
+  private function setActivitiesAsCompleted($absentDateDurations) {
+    $this->recreateSubactivities($absentDateDurations);
+
+    $activityStatus = empty($this->_activityStatuses) ? 
+      CRM_HRAbsence_BAO_HRAbsenceType::getActivityStatus('name') : 
+      $this->_activityStatuses;
+
+    return civicrm_api3('Activity', 'create', [
+      'sequential' => 1,
+      'id' => $this->_activityId,
+      'activity_type_id' => $this->_activityTypeID,
+      'status_id' => CRM_Utils_Array::key('Completed', $activityStatus)
+    ]);
+  }
+
+  /**
+   * Deletes subactivities of leave request and creates them again for each day.
+   * 
+   * @param array $absentDateDurations
+   *   List of durations (days) of leave
+   */
+  private function recreateSubactivities($absentDateDurations) {
```

Comment:
please stick to camelCase .. **recreateSubactivities** should be **recreateSubActivities**